torch/nn/modules/container.py

```diff
@@ -115,27 +130,38 @@ def __iter__(self):
     def __iadd__(self, modules):
         return self.extend(modules)
 
-    def append(self, module):
-        r"""Appends a given module at the end of the list.
+    def append(self, module, prefix=None):
```

Comment:
I don't think having a `prefix` arugment is necessary. It will end up calling the module like "prefix.9", which doesn't really help. It's not a valid identifier in Python, and we usually use dots to denote that there's a module called "prefix" and it has a submodule named "9" (so it's a nested module structure). I'd rather remove it from this diff.

Reply:
Yea, this is something I thought about for a while.  Right now the keys of the OrderedDict are used to maintain the ordering and thus other key names cannot be stored.  I used the data type structure itself to maintain the order and stored the original key values if available.  But this leads to the problem of possible duplicate keys, but only when using the named_modules/parameters functions. 

I added the prefix argument because if one adds more than a single set of named_modules(), often times there will be a module named "0" or "1" and that would overwrite any other previous module with those key values.

So regarding this I have two ideas:

1) I could remove the prefix argument, but that will cause funny behavior if one adds more than one named_modules/parameters.

2) I could remove the prefix argument AND the code for accepting named_modules/parameters() which loses the ability to maintain key names.

Open to other ideas too

Reply:
3) remove the prefix argument and allow named_*() but not use the key name and use an index like the modules() and parameters() functions